src/server/streamableHttp.ts

```diff
@@ -463,7 +463,7 @@ export class StreamableHTTPServerTransport implements Transport {
    * Returns true if the session is valid, false otherwise
    */
   private validateSession(req: IncomingMessage, res: ServerResponse): boolean {
-    if (!this._initialized) {
+    if (this.sessionId && !this._initialized) {
```

Comment:
```suggestion
    if (this.sessionId === undefined) {
      // If the session ID is not set, the session management is disabled
      // and we don't need to validate the session ID
      return true;
    }
    if (!this._initialized) {
```

Reply:
This would make more sense. Moving this check for stateless to the top of the method makes it obvious that for the rest, we're dealing with a stateful connection
.